### DISCUSSION TITLE
HIP-365 Partially Fungible Tokens

### DISCUSSION THREAD

**Discussion by superboo** [hiero-ledger/hiero-improvement-proposals#384] [Edits]
Discussion for [#365](https://github.com/hashgraph/hedera-improvement-proposal/pull/365)

**Comment by rocketmay** [comment]
In the _Introduction of Sub-Token Type_ section, the TokenCreateTransactionBody JSON seems to be related to Fungible_Partial_Abstract and not Fungible_Partial_Derived.

Transfer Policies - Could examples be provided of a simple and complex transfer policy?

Get Sub-Token Balances: How are sub-tokens tracked by the root token? How does retrieving all the sub-tokens associated with a root token work?


Edge case thoughts:
- Are there any issues if a FUNGIBLE_PARTIAL_DERIVED token is reassigned to a different root token?
- Are there security measures are in place to prevent an entity from creating a sub-token for a root token that they do not own?
- What happens to sub-tokens if a root token is deleted?

Imagine a scenario with root token **R**, and sub tokens of R named **A** and **B**.
- If an account is associated to **R** but not A and B, does a transfer of **A** fail?
- If an account is associated to R and A but not B, does a transfer of B fail? If so, this means that every account that associates to R must also pay for association with every sub-token as well. In addition, if a sub-token **C** is added, how are accounts updated (also paying the association fee)?

Given the above edge cases, perhaps it would it be better for partially-fungible tokens to instead be 'serial numbers' underneath a collection ID, similar to NFT's? Each serial number could have its own balance, rules, tags, admin/supply/etc keys.
